### ISSUE TITLE
[Feature request]: Custom color for unused monitor zones

### ISSUE THREAD

**Issue by davidhedlund** [libretro/RetroArch#16169]
## Description

Since the background color for unused zones of the monitor in RA is black, it becomes more difficult to predict what comes up in games with dark backgrounds. For example, in Alien 3 for SNES.

### Feature request

It would be useful with a new setting that made it possible to set a custom background color for the unused zones of the monitor in RA. Here's an example with gray background colors, as you can see it's easier to focus on the end of the screen in the game, rather to stare into the darkness:

![image](https://github.com/libretro/RetroArch/assets/6416073/a35ea7ed-99ad-4bf0-974e-ffd47de69efc)

### Actual behavior

[What is actually happening]

### Steps to reproduce the issue

Run around in Alien 3 for SNES to get an impression.

![image](https://github.com/libretro/RetroArch/assets/6416073/d09344a2-7f1e-4cca-ab85-a8a86a7ed444)

**Comment by davidhedlund** [comment]
Feel free to change the topic of this issue. I don't know what this zone is called.

**Comment by warmenhoven** [comment]
When there are bars on the sides of the image it's known as pillboxing. If it's on the top and bottom it's called letterboxing. And if it's on all four sides it's windowboxing.

**Comment by davidhedlund** [comment]
> When there are bars on the sides of the image it's known as pillboxing. If it's on the top and bottom it's called letterboxing. And if it's on all four sides it's windowboxing.

Thank you, I added "bars on the sides of the image" to include everything.

**Comment by zoltanvb** [comment]
I spent some time looking at this, but the result is that "proper implementation" (adding it as a generally available option in video settings) could get quite complex with lots of side effects, for the following reasons:
- the place where it could probably be injected is a central part of the drivers, the main frame render routine ([example from glcore](https://github.com/libretro/RetroArch/blob/fb63c79f32b916633f07c8a04679dd428e621737/gfx/drivers/gl3.c#L2591))
- this is driver dependent, so quite a few would need to be covered and tested
- there could be all sorts of side effects, for example when BFI is active, or with shaders, or with HW rendering active, even menu background can have a strange look

All in all, me personally would rather not change this. Could similar use case be achieved by bezels (overlays or shaders)?

On the other hand: for specific personal use, it is probably quite easy to change that one line (maybe a few), for that one driver, and have a patched RA executable with a fixed framebuffer background color.

**Comment by hizzlekizzle** [comment]
yes, we have some shaders that can handle this sort of thing (currently, none of them specifically change the color statically, but that would be easy enough to implement) and that's typically how we deal with this sort of thing. Having a general setting for it in RetroArch would be great, but it does sound like a real can of worms, since so much depends on glClearColor.
